### ISSUE TITLE
Swift 4

### ISSUE THREAD

**Issue by neoneye** [raulriera/Bike-Compass#15]
Your [blog post about animations](https://medium.com/@raulriera/custom-view-controller-presentations-the-right-way-53e8e8e8118b) is pretty cool.

However I'm unable to compile using Xcode9.2.  I also tried using Xcode9.0.1, but no luck.

Your code looks good.

**Comment by raulriera** [comment]
Hi,

Thanks! Yeah, given that is Swift. It needs to be updated for the latest versión if you wish to use the latest Xcode

**Comment by lolgear** [comment]
@raulriera 
Yes, it needs to be updated to Swift 4.2.
Could you do it?

I am bit confused with Decodable framework and its operators ( oh, swift evolution should prevent all these useless frameworks, heh ).

**Comment by esikmalazman** [comment]
Hi @raulriera, can I contribute to maintain the project?

**Comment by raulriera** [comment]
Hey, you sure can :)

**Comment by esikmalazman** [comment]
Perfect :)

**Comment by esikmalazman** [comment]
I want to update the package dependencies to SPM but some of the libraries require iOS 12. Can I bump the project to iOS 12 or 13 @raulriera ?

**Comment by raulriera** [comment]
Yeah, feel free to make any update you want (as long as the resulting PR builds and produces a working app)
